forem/forem
# Cleaning Test Runs for End to End (E2E) Tests

## What type of PR is this? (check all applicable)

- [x] Testing infrastructure

## Description

Seeds the E2E test server with data and wipes any changes made between test runs. There are 25 file changes, but there are deletes in there. I could break this up to first add the cypress-rails gem, and pull out a few other things, but I think it makes sense to have this all together.

## Related Tickets & Documents

#382

## QA Instructions, Screenshots, Recordings

End to End Tests should pass on CI.

1. To test locally, pull this branch down and make sure everything is up to date by running `bin/setup`
2. From the command line, run `yarn e2e`. You will be prompted to set up the end to end (E2E) test database. Type `y` or `Y` to install the E2E test database.

```bash
yarn run v1.22.5
$ bin/e2e
Do you need to set up your end to end (E2E) testing database? Answer yes if this is your first time running E2E tests on your local machine or you need to recreate your E2E test database. (y/n) 
```

The E2E test database will install.

```bash
Setting up the E2E database before running E2E tests...


Dropped database 'PracticalDeveloper_test'
Created database 'PracticalDeveloper_test'

...
```

3. The Cypress test runner will open.

![image](https://user-images.githubusercontent.com/833231/105519345-4dd48800-5ca7-11eb-9b72-91db60a10454.png)

4. Press the `➠a Run Integration link on the right near the top of the test runner.

![image](https://user-images.githubusercontent.com/833231/105519402-604ec180-5ca7-11eb-8c95-38b92faea3e5.png)

5. All the tests should run and pass.

![image](https://user-images.githubusercontent.com/833231/105519505-7fe5ea00-5ca7-11eb-8ae7-72cf0145024e.png)

**Note: The e2e application running uses the test database and not your development database so no data loss will occur for your local development environment.**


### UI accessibility concerns?

N/A. This is testing infrastructure.

## Added tests?

      have not been included_

## Added to documentation?

In regards to documentation, I will do this in a separate PR as this PR is already getting a little large and I'd like to think about the documentation a bit first.

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![The Big Lebowski tapping the top of the inside of his car looking very happy](https://media.giphy.com/media/tyxovVLbfZdok/giphy.gif)
